### PR TITLE
PHP 8: Fix major compatibility issues

### PR DESCRIPTION
**Pull request by jbeyerstedt**
PHP8 deprecated, that class member variables can be created outside the class definition or constructor, which prevented the code to run at all. Additionally the error handling has changed, which has lead to multiple other errors during the runtime.

Finally, strftime was deprecated in PHP 8.1.

### CODE REVIEW

**Comment by jbeyerstedt** [comment]
Don't know, if this breaks anything in php 7.4 (or earlier) as I don't have old php on my laptop any more.

**Comment by jbeyerstedt** [comment]
Works with PHP8 on my machine and seems to work with the PHP7 docker container mentioned in the readme.

But there seems an issue to load player.js as it has a query param specifying the version number. But this also happens with the current master branch (with PHP in docker).

**Comment by jbeyerstedt** [comment]
> But there seems an issue to load player.js as it has a query param specifying the version number. But this also happens with the current master branch (with PHP in docker).

This should not be an issue as the assets are usually served by nginx directly. At least that the way how the LBs are doing it.

File: model/ConferenceJson.php [hunk 1]
```diff
@@ -12,26 +12,32 @@ public function __construct($json, $mandator)
...
 
 		$this->rooms = [];
-		$rooms = (is_array(@$c->rooms) ? $c->rooms : @$c->rooms->nodes) ?: [];
+		if (isset($c->rooms)) {
+			if (is_array($c->rooms)) {
+				$rooms = $c->rooms;
+			} else {
+				$rooms = isset($c->rooms->nodes) ? $c->rooms->nodes : [];
+			}
+		}
```
Comment (saerdnaer) [inline model/ConferenceJson.php:24]: It was so much shorter before 🙈

**Review by saerdnaer — APPROVED** [review]
LGTM

File: model/ConferenceJson.php (hunk 1 above)
Comment (jbeyerstedt) [reply model/ConferenceJson.php:24]: But I haven't found a shorter version which works and is readable.

**Pull request merged**